src/Filter/AbstractFilter.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber\Filter;
+
+use Laminas\Filter\FilterInterface;
+use Laminas\I18n\PhoneNumber\CountryCode;
+use Laminas\I18n\PhoneNumber\Exception\ExceptionInterface;
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+
+use function is_scalar;
+
+abstract class AbstractFilter implements FilterInterface
+{
+    private CountryCode $countryCode;
+
+    final public function __construct(CountryCode $countryCode)
+    {
+        $this->countryCode = $countryCode;
+    }
+
+    /** @param mixed $value */
+    protected function mixedToPhoneNumber($value): ?PhoneNumberValue
+    {
+        if ($value instanceof PhoneNumberValue) {
+            return $value;
+        }
+
+        if (! is_scalar($value)) {
+            return null;
+        }
+
+        $input = (string) $value;
```

Comment:
What if it is an `object` that cannot be cast to `string`? :thinking:

Reply:
It's already been checked as a scalar